File: apps/api/internal/model/prediction.go
package model

type ShapFactor struct {
	Feature string  `json:"feature"`
	Value   float64 `json:"value"`
	Impact  string  `json:"impact"`
}

type PredictionResponse struct {
	HomeWinProb           float64      `json:"home_win_prob"`
	DrawProb              float64      `json:"draw_prob"`
	AwayWinProb           float64      `json:"away_win_prob"`
	HomeQualificationProb *float64     `json:"home_qualification_prob,omitempty"` // <--- Harus pakai pointer *float64
	AwayQualificationProb *float64     `json:"away_qualification_prob,omitempty"` // <--- Harus pakai pointer *float64
	AIAnalysis            string       `json:"ai_analysis"`
	TopFactors            []ShapFactor `json:"top_factors"`
}

File: apps/ai-service/app/schemas/prediction.py
from pydantic import BaseModel
from typing import Optional

class PredictionRequest(BaseModel):
    home_team: str
    away_team: str
    match_leg: int
    home_leg1_score: Optional[int] = None
    away_leg1_score: Optional[int] = None

class PredictionResponse(BaseModel):
    home_win_prob: float
    draw_prob: float
    away_win_prob: float
    home_qualification_prob: Optional[float] = None
    away_qualification_prob: Optional[float] = None
    ai_analysis: str